Microsoft.HealthVault.Fhir/Transformers/ObservationToHealthVault.cs

```diff
@@ -32,39 +35,97 @@ public static class ObservationToHealthVault
         /// <returns>The HealthVault thing</returns>
         public static ThingBase ToHealthVault(this Observation observation)
         {
-            // ToDo: detect the type from the codeable values
-            return observation.ToWeight();
+            return observation.ToHealthVault(VocabToHealthVaultHelper.DetectHealthVaultTypeFromObservation(observation));
         }
 
-        private static Weight ToWeight(this Observation observation)
+        private static ThingBase ToHealthVault(this Observation observation, Type type)
         {
-            var weight = new Weight();
-            var value = observation.Value as Quantity;
+            if (type == typeof(Weight))
+            {
+                return observation.ToWeight();
+            }
```

Comment:
Yes, I agree .. it could be driven from a dictionary .. <T, Func<>> no big if statement block .. 

---
In reply to: [124273929](https://github.com/Microsoft/healthvault-fhir-library/pull/3#discussion_r124273929) [](ancestors = 124273929)